src/components/NewEventForm/NewEventForm.tsx

```diff
@@ -0,0 +1,218 @@
+import React from 'react';
+import { Formik, Form } from 'formik';
+import { addDays, format, set, startOfDay } from 'date-fns';
+import * as Yup from 'yup';
+
+import Button from '../inputs/Button';
+import FormikInputField from '../FormikFields/FormikInputField/FormikInputField';
+import FormikTextAreaField from '../FormikFields/FormikTextAreaField/FormikTextAreaField';
+import FormikSelectorField from '../FormikFields/FormikSelectorField/FormikSelectorField';
+import {EventFormWrapper} from './styled';
+import { EVENT_STATUS_OPTIONS } from 'src/utils/constants';
+import { EVENT_OPTIONS } from '../EventForm/constants';
+
+export type EventDataInput = {
+  id?: string;
+  name?: string;
+  description?: string;
+  startTime?: Date;
+  endTime?: Date;
+  imageUrl?: string;
+  imageAlt?: string;
+  location?: string;
+  eventType?: string;
+  eventStatus?: string;
+  teamsUrl?: string;
+};
+
+export type EventDataOutput = {
+  id?: string;
+  name: string;
+  description: string;
+  startTime: Date;
+  endTime: Date;
+  imageUrl?: string;
+  imageAlt?: string;
+  location?: string;
+  eventType: string;
+  eventStatus: string;
+  teamsUrl?: string;
+};
+
+type EventFormProps = {
+  headerTitle: string;
+  submitTitle: string;
+  onSubmit: (data: EventDataOutput) => Promise<void>;
+} & EventDataInput;
+
+function NewEventForm({ onSubmit, ...rest }: EventFormProps) {
+
+  // Declartions
+  const now = new Date();
+  const tomorrow = addDays(startOfDay(now), 1);
+
+  // Utilities
+  const setHours = (date: Date, hours: number) =>
+    set(date, { hours: hours, minutes: 0, seconds: 0, milliseconds: 0 });
+  const parseDate = (date: string, time: string) => new Date(date + 'T' + time);
+
+  // Functions
+  const eventDate = format(tomorrow, 'yyyy-MM-dd');
+  const eventStartTime = format(setHours(tomorrow, 16), 'HH:mm');
+  const eventEndTime = format(setHours(tomorrow, 17), 'HH:mm');
+
+
+  // Initiation of values
+  const InitialValues = { 
```

Comment:
This name shouldn't be in PascalCase:
```suggestion
  const initialValues = { 
```